File: tools/dbench/answers/go_sonnet/s3/e2_missing_file.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func parseJSONFile(path string) (map[string]interface{}, error) {
	data, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}
	var m map[string]interface{}
	if err := json.Unmarshal(data, &m); err != nil {
		return nil, err
	}
	return m, nil
}

func main() {
	name := "default"
	if m, err := parseJSONFile("missing.cfg"); err == nil {
		if v, ok := m["name"].(string); ok {
			name = v
		}
	}
	fmt.Println(name)

	port := 0
	if m, err := parseJSONFile("app.cfg"); err == nil {
		if v, ok := m["port"].(float64); ok {
			port = int(v)
		}
	}
	fmt.Println(port)
}

File: tools/dbench/answers/python_haiku/s2/e2_missing_file.py
import json
name = "default"
try:
    f = open("missing.cfg")
    name = json.load(f).get("name", "default")
    f.close()
except Exception:
    pass
print(name)
port = 0
try:
    f = open("app.cfg")
    port = json.load(f).get("port", 0)
    f.close()
except Exception:
    pass
print(port)
